lib/galaxy/tools/parameters/basic.py

```diff
@@ -1012,18 +1025,22 @@ def from_json(self, value, trans, other_values=None, require_legal_value=True):
             if is_runtime_value(value):
                 return None
             if value in legal_values:
-                return value
+                rval = value
             elif value in fallback_values:
-                return fallback_values[value]
+                rval = fallback_values[value]
             elif not require_legal_value:
-                return value
+                rval = value
             else:
                 raise ParameterValueError(
                     f"an invalid option ({value!r}) was selected (valid options: {','.join(legal_values)})",
                     self.name,
                     value,
                     is_dynamic=self.is_dynamic,
                 )
+            if self.multiple:
+                return [rval]
```

Comment:
Maybe use `listify` so you're not returning a list of `None` or a nested list ? I think type annotations would also help a lot here, and you could possibly break this up into getting the legal values, then applying the conditional logic around multiple and optional.

Reply:
> I'd like to see the results against tools-iuc as well, this is one of these super-risky changes IMO.

I completely agree. Might be complicated to interpret the result given that more than 200 tests fail/error at the moment. No idea how to improve situation there .. all my efforts to make all tests pass seem insufficient.

I will try to start one to see already now what's the outcome. Then we can also compare to the results when I implemented the changes suggested by you other comment. 


> Maybe use listify so you're not returning a list of None or a nested list ? 

Good idea. 

> I think type annotations would also help a lot here, 

Indeed .. like always :) Lets see if I get this sorted out. 

> and you could possibly break this up into getting the legal values, then applying the conditional logic around multiple and optional.

Thanks for the nudge. I though about it, but wanted to keep the diff small .. will try to come up with something better.

Reply:
Just noticed that this problem only occurs during testing. If I execute the test in `planemo serve` everything is fine. 

So the question is really if my change is just fixing symptoms...

Do you know by any chance where the `value` that the `from_json` function gets is generated? Must be something in parsing tests from xml or in `tool_util/verify` -- I guess...